Cms/ProfileBioEdit.ascx.cs

```diff
@@ -0,0 +1,2389 @@
+            public const string ContactFields = "Phone,Email,EmailPreference,CommunicationPreference";
+        }
+
+        private static class AttributeCategory
+        {
+            public const string PersonFields = "Person Fields";
+            public const string ContactFields = "Contact Fields";
+            public const string Campus = "Campus";
+            public const string Attributes = "Attributes";
+        }
+
+        private List<PersonFamilyMember> FamilyMembers
+        {
+            get { return Session["FamilyMembers"] as List<PersonFamilyMember>; }
+            set { Session["FamilyMembers"] = value; }
+        }
+
+        #region Properties
+
+
+        #endregion Properties
+
+        #region Base Control Methods
+
+        /// <summary>
+        /// Raises the <see cref="E:System.Web.UI.Control.Init" /> event.
+        /// </summary>
+        /// <param name="e">An <see cref="T:System.EventArgs" /> object that contains the event data.</param>
+        protected override void OnInit( EventArgs e )
+        {
+            base.OnInit( e );
+
+            this.BlockUpdated += Block_BlockUpdated;
+            this.AddConfigurationUpdateTrigger( upnlCareEntry );
+
+            BuildForm();
+        }
+
+        /// <summary>
+        /// Raises the <see cref="E:System.Web.UI.Control.Load" /> event.
+        /// </summary>
+        /// <param name="e">The <see cref="T:System.EventArgs" /> object that contains the event data.</param>
+        protected override void OnLoad( EventArgs e )
+        {
+            base.OnLoad( e );
+
+            //confirmExit.Enabled = true;
+
+            lbSave.ValidationGroup = BlockValidationGroup;
+
+            lbCancel.ValidationGroup = BlockValidationGroup;
+            lbCancel.CausesValidation = false;
+
+            var displayButtonsInPanels = GetAttributeValue( AttributeKey.DisplayButtonsInAllPanels ).AsBoolean();
+
+            lbSave.Visible = !displayButtonsInPanels;
+            lbCancel.Visible = !displayButtonsInPanels;
+
+            pnlProfilePanels.DefaultButton = lbSave.ID;
+
+            // Initialize year picker javascript per grade ddl
+            //if ( ddlGrade.Visible )
+            //{
+            //    //ScriptManager.RegisterStartupScript( ddlGrade, ddlGrade.GetType(), "grade-selection-" + BlockId.ToString(), ddlGrade.GetJavascriptForYearPicker( ypGraduation ), true );
+            //}
+
+
+        }
+
+        #endregion Base Control Methods
+
+        #region Events
+
+        /// <summary>
+        /// Handles the BlockUpdated event of the control.
+        /// </summary>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
+        protected void Block_BlockUpdated( object sender, EventArgs e )
+        {
+            BuildForm();
```

Comment:
Should this be changed to a modal to tell user to refresh the page if we can't get this to work?